### PR TITLE
Request uri

### PR DESCRIPTION
**Pull request by medjedovicm**

### CODE REVIEW

File: src/SASViyaApiClient.ts [hunk 1]
```diff
@@ -280,22 +293,41 @@ export class SASViyaApiClient {
   /**
    * Creates a file in the specified folder.
```
Comment (allanbowe) [inline src/SASViyaApiClient.ts:294]: this should say "Creates a job"

File: src/SASViyaApiClient.ts [hunk 2]
```diff
@@ -14,10 +14,11 @@ import { Job, Session, Context, Folder } from "./types";
  *
  */
 export class SASViyaApiClient {
+  private rootFolderMap = new Map<string, Job[]>();
+  
   constructor(
     private serverUrl: string,
-    private rootFolderName: string,
-    private rootFolderMap = new Map<string, Job[]>()
```
Comment (saadjutt01) [inline src/SASViyaApiClient.ts]: these 2 are gets added after merge.

File: src/SASjs.ts [hunk 3]
```diff
@@ -841,13 +841,13 @@ export default class SASjs {
         ? "/SASLogon/logout?"
         : "/SASLogon/logout.do?";
 
-    if (this.sasjsConfig.serverType === ServerType.SASViya) {
+    if (this.sasjsConfig.serverType === ServerType.SASViya && !this.sasViyaApiClient) {
```
Comment (saadjutt01) [inline src/SASjs.ts:844]: if we are preventing creation of new client, then how will sasViyaApiClient wil have updated serverUrl or AppLoc?

File: src/SASjs.ts [hunk 4]
```diff
@@ -59,7 +59,7 @@ export default class SASjs {
     };
 
     this.setupConfiguration();
-    this.sasViyaApiClient!.populateJobMap();
+    this.sasViyaApiClient!.populateRootFolderMap();
```
Comment (saadjutt01) [inline src/SASjs.ts:62]: typescript gives error on build 
_Property 'populateRootFolderMap' is private and only accessible within class 'SASViyaApiClient'._

**Comment by krishna-acondy** [comment]
Closing as changes have been manually merged into https://github.com/macropeople/sasjs/pull/185.

**Pull request closed without merging**